api/ratelimit/v1alpha1/ratelimit_types.go

```diff
@@ -0,0 +1,149 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+
+// Ratelimit allows the user to limit the number of incoming requests
+// to a predefined value based on attributes within the traffic flow.
+type Ratelimit struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of HTTPRoute.
+	Spec RatelimitSpec `json:"spec"`
+
+	// Status defines the current state of Ratelimit.
+	Status RatelimitStatus `json:"status,omitempty"`
```

Comment:
Should this API type surface status or should referents be responsible for surfacing status?

Reply:
good point, if its a filter extension, a status field is unneeded, if not, it probably is needed

Reply:
removed `status` for now, since the initial design will be leveraging the filter extension